Packages/vcs/Lib/Canvas.py

```diff
@@ -2759,16 +2764,20 @@ def __plot (self, arglist, keyargs):
                                     'ext_2',
                                     'missing']:
                             setattr(copy_mthd,att,getattr(m,att))
-        elif arglist[0] is not None and arglist[0].rank()<2 and arglist[3] in ['boxfill','default'] and not isinstance(arglist[0].getGrid(),cdms2.gengrid.AbstractGenericGrid):
+        elif arglist[0] is not None and arglist[0].rank()<2 and arglist[3] in ['boxfill','default'] and not isinstance(g0,cdms2.gengrid.AbstractGenericGrid):
             arglist[3]='1d'
             try:
                 tmp=self.getyxvsx(arglist[4])
                 #tmp.list()
             except Exception,err:
                 arglist[4]='default'
+        elif g0 is not None and (arglist[0] is not None and isinstance(arglist[0],cdms2.avariable.AbstractVariable) and not isinstance(g0,cdms2.grid.AbstractRectGrid)) and arglist[3] in ["boxfill","default"] and arglist[4]=="default":
+          arglist[3]="meshfill"
 
 ##                         arglist[4]=copy_mthd.name
-
+        # Ok let's check for meshfill needed
+        if g0 is not None and (arglist[0] is not None and isinstance(arglist[0],cdms2.avariable.AbstractVariable) and not isinstance(arglist[0].getGrid(),cdms2.grid.AbstractRectGrid)) and arglist[3] not in ["meshfill",]:
+          raise RuntimeError("You are attempting to plot unstructured grid with a method that is not meshfill")
```

Comment:
@ThomasMaxwell thanks for checking. Specifically that is the area I was worried about. Glad it didn't break your stuff.